### PR TITLE
Fix some warnings in mlas

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

1. C26451. Arithmetic overflow: Using operator 'operator' on a size-a byte value and then casting the result to a size-b byte value. Cast the value to the wider type before calling operator 'operator' to avoid overflow. Ignore some occurrences.
2. Use constexpr to replace const.
3. Use "[[fallthrough]]" when we fall through from one label to another. 

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/mlas/lib/qgemm.h [hunk 1]
```diff
@@ -73,7 +73,7 @@ MlasGemmQuantTryGemvKernel(
 }
 
 template <typename KernelType>
-MLAS_FORCEINLINE
+MLAS_FORCEINLINE constexpr
```
Comment (yufenglee) [inline onnxruntime/core/mlas/lib/qgemm.h:76]: any reason to add constexpr here? ZeroPointA  is not known at compiling time

Comment (yufenglee) [reply onnxruntime/core/mlas/lib/qgemm.h:76]: similar to other functions

Comment (snnn) [reply onnxruntime/core/mlas/lib/qgemm.h:76]: Here constexpr means if the arguments are known at compile time, the result would be too.

Comment (yufenglee) [reply onnxruntime/core/mlas/lib/qgemm.h:76]: yes, but as i mentioned, ZeroPointA is not known at compiling time. It is meaningless. If it is for some style enforcement, that's fine to have it.

**Review by yufenglee — APPROVED** [review]

**Pull request merged**